fs-storage/src/file_storage.rs

```diff
@@ -176,9 +157,77 @@ where
                 ),
             ));
         }
+
+        Ok(data)
+    }
+}
+
+impl<K, V> BaseStorage<K, V> for FileStorage<K, V>
+where
+    K: Ord
+        + Clone
+        + serde::Serialize
+        + serde::de::DeserializeOwned
+        + std::str::FromStr,
+    V: Clone
+        + serde::Serialize
+        + serde::de::DeserializeOwned
+        + std::str::FromStr
+        + Monoid<V>,
+{
+    /// Set a key-value pair in the storage
+    fn set(&mut self, key: K, value: V) {
+        self.data.entries.insert(key, value);
+        self.modified = std::time::SystemTime::now();
+    }
```

Comment:
Why not write to the file system whenever a user sets a value?

Reply:
We've discussed this in the other PR. The idea is to allow the user to control when exactly the write is happening. Performance reasons mostly. Storage values can be updated often, but writing to disk is a slow operation. Also, we run coroutines in Kotlin to update storages. It's difficult to do sequentially because it's IO and we don't want to freeze UI thread. So, if we create lots of coroutines they start race with each other. We have atomic versions exactly for this reason, but it's also nice to just reduce the load.

Reply:
Thanks for clarifying. I think I had forgotten about this.